students/shirin_ak/Session04/mailroom.py

```diff
@@ -0,0 +1,225 @@
+#Convert the amount into a number 
+#Add that amount to the donation history of the selected user.
+#Finally, use string formatting to compose an email thanking the donor for their generous donation. Print the email to the terminal and return to the original prompt.
+
+
+
+
+Donors = {'william gates':[3, 3000.45], 
+               'Jeff Bezos': [2, 3452.00], 
+               'Paul Allen': [2, 9970.65], 
+               'Mark Zuckerberg': [5,5000.75 ], 
+               'David': [3, 2500.17], 
+               'Michael': [4, 1500.00], 
+               'Allen': [1, 2400.35]}
+
+
+def Create_report(): 
+
+    """Create a detailed report of donors and their history of the amounts they have donated"""
+    
+    donor_report = []
+    for name in Donors: 
+        total_donation = (Donors[name][1]) 
+ 
+        donation_times = (Donors[name][0]) 
+        avg_donate = round((total_donation / donation_times), 2) 
+        donor_report.append((name, total_donation, donation_times, avg_donate)) 
+        donor_report.sort(key=lambda x: x[1] , reverse=True) 
+
+    print("\n\n") 
+    print("{:16} {:20} {:25} {:25}\n".format( "Name", "Total Donation", "Number of Gifts", "Average Gifts")) 
+    for i in donor_report: 
+        print("{:16} {:<20} {:<25} {:<25}".format( i[0], i[1], i[2], i[3])) 
+
+    print("\n\n\n")
+
+
+
+def Add_donor():
+    
+    """ prompt for a Full Name.if user types 'list' show donor names,if types 'quit'
+        return to the main menu ,if types a name not in the list add that to the list,
+        if types a name in the list use it and prompt for donation amount,Add that amount
+        to the donation history """
+
+    while True: 
+
+        print("\nEnter 'list' to see list of names.") 
+
+        print("Enter 'quit' to go back.")
+        print("Enter 'name' to add new donor.")
+        
+        name = input("Please enter a name: ")
+        
+        if (name == 'list'):
+ 
+            Show_list()
+            
+        elif (name == 'quit'): 
+            break
+        
+        else: 
+
+            donation_amount = get_donation() 
+
+            if (donation_amount == 'quit'):
+ 
+                break
+            
+            if (name not in Donors):
+                
+                Donors[name] = [1, donation_amount]
+                
+            else:
+                
+                Donors[name][0] = Donors[name][0] + 1 
+ 
+                Donors[name][1] = Donors[name][1] + donation_amount
+ 
+            Send_Thank_you_letter(name, donation_amount)
+ 
+            break
+        
+        
+def Show_list():
+    
+    print("\nDonor Names:") 
+    print("-" *15 ) 
+    for name in Donors:
+ 
+        print(name)       
+
+
+def get_donation():
+    """ Take donation from the donor"""
+
+    while True: 
+        print("\nPlease enter donation amount. " 
+
+             " Or Enter 'quit' to return to main menu.")
+ 
+        donation = input("$")
```

Comment:
You have this single input statement broken out into 3 lines when it could be one
```
donation = input("\nPlease enter a donation amount.\nOr enter 'quit' to return to main menu.\n$")
```